Description: Avoid >& bashism in tests.
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Forwarded: https://gitlab.com/gnutls/gnutls/-/merge_requests/1518
Last-Update: 2022-01-21

--- gnutls28-3.7.3.orig/tests/protocol-set-allowlist.sh
+++ gnutls28-3.7.3/tests/protocol-set-allowlist.sh
@@ -116,7 +116,7 @@ with_config_file() {
 
 # Smoke --list, @SYSTEM
 
-with_config_file "${CLI}" --list -d 4 --priority @SYSTEM &>"${TMPFILE_LIST}"
+with_config_file "${CLI}" --list -d 4 --priority @SYSTEM > "${TMPFILE_LIST}" 2>&1
 if test $? != 0; then
 	${CAT} "${TMPFILE_LIST}"
 	echo 'fails with just @SYSTEM'
@@ -131,7 +131,7 @@ fi
 # Smoke-test that TLS 1.3 is enableable with these algorithms
 
 with_config_file \
-	"${CLI}" --list -d 4 --priority @SYSTEM:+VERS-TLS1.3 &>"${TMPFILE_LIST}"
+	"${CLI}" --list -d 4 --priority @SYSTEM:+VERS-TLS1.3 > "${TMPFILE_LIST}" 2>&1
 if test $? != 0; then
 	${CAT} "${TMPFILE_LIST}"
 	echo 'listing algorithms fails with @SYSTEM:+VERS-TLS1.3'
@@ -146,7 +146,7 @@ fi
 # Smoke-test that TLS 1.1 is enableable with these algorithms
 
 with_config_file \
-	"${CLI}" --list -d 4 --priority @SYSTEM:+VERS-TLS1.1 &>"${TMPFILE_LIST}"
+	"${CLI}" --list -d 4 --priority @SYSTEM:+VERS-TLS1.1 > "${TMPFILE_LIST}" 2>&1
 if test $? != 0; then
 	${CAT} "${TMPFILE_LIST}"
 	echo 'listing algorithms fails with @SYSTEM:+VERS-TLS1.1'
--- gnutls28-3.7.3.orig/tests/system-override-special-allowlist.sh
+++ gnutls28-3.7.3/tests/system-override-special-allowlist.sh
@@ -71,7 +71,7 @@ export GNUTLS_SYSTEM_PRIORITY_FAIL_ON_IN
 
 # Smoke --list, @SYSTEM
 
-${CLI} --list -d 4 --priority @SYSTEM &>"${TMPOUTFILE}"
+${CLI} --list -d 4 --priority @SYSTEM > "${TMPOUTFILE}" 2>&1
 if test $? != 0; then
 	cat "${TMPOUTFILE}"
 	echo 'fails with just @SYSTEM'
@@ -97,7 +97,7 @@ ${SED} 's/for @SYSTEM/for ---PRIORITY---
 
 # Smoke-test a no-op %STATELESS_COMPRESSION, expect --list to stay the same
 
-${CLI} --list -d 4 --priority @SYSTEM:%STATELESS_COMPRESSION &>"${TMPOUTFILE}"
+${CLI} --list -d 4 --priority @SYSTEM:%STATELESS_COMPRESSION > "${TMPOUTFILE}" 2>&1
 if test $? != 0; then
 	cat "${TMPOUTFILE}"
 	echo 'fails with %STATELESS_COMPRESSION'
@@ -112,7 +112,7 @@ fi
 
 # Smoke-test %NONEXISTING_OPTION, expect a syntax error
 
-${CLI} --list -d 4 --priority @SYSTEM:%NONEXISTING_OPTION &>"${TMPOUTFILE}"
+${CLI} --list -d 4 --priority @SYSTEM:%NONEXISTING_OPTION > "${TMPOUTFILE}" 2>&1
 if test $? = 0; then
 	cat "${TMPOUTFILE}"
 	echo 'unknown option was not caught'
@@ -132,7 +132,7 @@ while read special; do
 		continue  # see below
 	fi
 	prio="@SYSTEM:%$special"
-	${CLI} --list -d 4 --priority "$prio" &>"${TMPOUTFILE}"
+	${CLI} --list -d 4 --priority "$prio" > "${TMPOUTFILE}" 2>&1
 	if test $? != 0; then
 		cat "${TMPOUTFILE}"
 		echo "fails with $prio"
@@ -148,7 +148,7 @@ done < "${TMPSPECIAL}"
 
 # Check that %NO_EXTENSIONS changes the output, capping it to TLS 1.2
 
-${CLI} --list -d 4 --priority @SYSTEM:%NO_EXTENSIONS &>"${TMPOUTFILE}"
+${CLI} --list -d 4 --priority @SYSTEM:%NO_EXTENSIONS > "${TMPOUTFILE}" 2>&1
 if test $? != 0; then
 	cat "${TMPOUTFILE}"
 	echo 'fails with just @SYSTEM'
